src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -2154,64 +2149,40 @@ static public TimeZoneInfo FindSystemTimeZoneById(string id)
             {
                 rawData = File.ReadAllBytes(timeZoneFilePath);
             }
-            catch (UnauthorizedAccessException e)
-            {
-                throw new SecurityException(Environment.GetResourceString("Security_CannotReadFileData", id), e);
-            }
-            catch (FileNotFoundException e)
+            catch (UnauthorizedAccessException ex)
             {
-                throw new TimeZoneNotFoundException(Environment.GetResourceString("TimeZoneNotFound_MissingData", id), e);
+                value = null;
```

Comment:
can you just initialize the value with null in the beginning and remove it from the rest.